diff --git a/app/controllers/tickets_controller.rb b/app/controllers/tickets_controller.rb
index 1e9408c..c3811a6 100644
--- a/app/controllers/tickets_controller.rb
+++ b/app/controllers/tickets_controller.rb
@@ -8,11 +8,12 @@ class TicketsController < ApplicationController
   def show; end
 
   def new
-    @ticket = Ticket.new(ticket_params)
+    @ticket = Ticket.new(ticket_params.merge(user: User.new))
   end
 
   def create
-    @ticket = Ticket.new(ticket_params)
+    user = User.find_or_initialize_by(ticket_params[:user_attributes])
+    @ticket = Ticket.new(ticket_params.except(:user_attributes).merge(user: user))
     if @ticket.save
       redirect_to @ticket
     else
@@ -27,7 +28,21 @@ class TicketsController < ApplicationController
   end
 
   def ticket_params
-    params.require(:ticket).permit(:name, :middle_name, :last_name, :passport, :train_id, :first_station_id,
-                                   :last_station_id, :user_id, :arrival, :departure, :user_id)
+    params
+      .require(:ticket)
+      .permit(
+        :name,
+        :middle_name,
+        :last_name,
+        :passport,
+        :train_id,
+        :first_station_id,
+        :last_station_id,
+        :arrival,
+        :departure,
+        user_attributes: [
+          :login,
+          :password
+                         ])
   end
 end
diff --git a/app/models/ticket.rb b/app/models/ticket.rb
index 6a8e449..c1c1ad2 100644
--- a/app/models/ticket.rb
+++ b/app/models/ticket.rb
@@ -4,4 +4,6 @@ class Ticket < ApplicationRecord
 
   belongs_to :first_station, class_name: 'RailwayStation', foreign_key: :first_station_id, inverse_of: :tickets
   belongs_to :last_station, class_name: 'RailwayStation', foreign_key: :last_station_id, inverse_of: :tickets
+
+  accepts_nested_attributes_for :user
 end
diff --git a/app/models/user.rb b/app/models/user.rb
index 6e5ef2c..a456b44 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -1,4 +1,3 @@
 class User < ApplicationRecord
-  has_many :tickets, dependent: :nullify, inverse_of: :user
-  accepts_nested_attributes_for :tickets
+  has_many :tickets, dependent: :nullify
 end
diff --git a/app/views/tickets/_form.html.erb b/app/views/tickets/_form.html.erb
index 0d8d205..3b7957e 100644
--- a/app/views/tickets/_form.html.erb
+++ b/app/views/tickets/_form.html.erb
@@ -1,25 +1,25 @@
-<%= form_with(model: @ticket, local: true, class: 'form-horizontal') do |form| %>
+<%= form_for(@ticket, class: 'form-horizontal') do |form| %>
   <%= render 'common/error', resource: @ticket %>
 
-  <div class="field">
-    <%= form.label :name %>
-    <%= form.text_field :name %>
-  </div>
+<!--  <div class="field">-->
+    <%#= form.label :name %>
+    <%#= form.text_field :name %>
+<!--  </div>-->
 
-  <div class="field">
-    <%= form.label :middle_name %>
-    <%= form.text_field :middle_name %>
-  </div>
+<!--  <div class="field">-->
+    <%#= form.label :middle_name %>
+    <%#= form.text_field :middle_name %>
+<!--  </div>-->
 
-  <div class="field">
-    <%= form.label :last_name %>
-    <%= form.text_field :last_name %>
-  </div>
+<!--  <div class="field">-->
+    <%#= form.label :last_name %>
+    <%#= form.text_field :last_name %>
+<!--  </div>-->
 
-  <div class="field">
-    <%= form.label :passport %>
-    <%= form.text_field :passport %>
-  </div>
+<!--  <div class="field">-->
+    <%#= form.label :passport %>
+    <%#= form.text_field :passport %>
+<!--  </div>-->
 
   <div class="field">
     <%= form.label :train %>
@@ -41,17 +41,23 @@
 
   <div class="field">
     <%= form.label :departure %>
-    <p><%= @ticket.departure.strftime('%H:%M') %></p>
-    <%= form.hidden_field(:departure, value: @ticket.departure) %>
+    <p><%= form.datetime_field :departure %></p>
   </div>
 
 
   <div class="field">
     <%= form.label :arrival %>
-    <p><%= @ticket.arrival.strftime('%H:%M') %></p>
-    <%= form.hidden_field(:arrival, value: @ticket.arrival) %>
+    <p><%= form.datetime_field :arrival %></p>
   </div>
 
+    <%= form.fields_for :user do |user_form|%>
+      <%= user_form.label :login %>
+      <%= user_form.text_field :login %>
+
+      <%= user_form.label :password %>
+      <%= user_form.password_field :password %>
+    <% end %>
+
   <div class="actions">
     <%= form.submit %>
   </div>
